### ISSUE TITLE
Feedback on Demo Performance

### ISSUE THREAD

**Issue by ToratEmetInWord** [majkimester/PdfiumViewer.WPF#2]
Hi there!

I noticed that the performance is a bit slow in the demo. It might be worth looking into that.

I believe your project has huge potential, and many people have been waiting for something like this for years.

Best of luck!

**Comment by majkimester** [comment]
Hi, thank you for your feedback. I know there is performance drawback compared to the Forms version of the same viewer. I tried to optimize it and I tried different method to render a  page and convert it to presentable image. I replaced the original solution with a better one but still not ideal. If you have any idea how to improve the performance let me know.

**Comment by pcinfogmach** [comment]
maybe this can help
https://weblog.west-wind.com/posts/2024/Jan/03/Working-around-the-WPF-ImageSource-Blues
I would like To point out that the viewer is heavily based on code instead of mvvm and xaml perhaps this can be a cause too.

**Comment by majkimester** [comment]
I read the article, which gives a hint for async loading the image.
Actually there is an easier way to do it with IsAsync binding, I tried it before but for me it gave me worse user experience. I was able to scroll quickly but the pages loaded asyc with a delay, so first blank pages later the page appearad. I decided to keep the blocking loading for now. 
The underlaying lib render the image which I need to convert to ImageSource to be able to present in WPF. I have no change to improve the rendering of the image by the lib, but I worked on different method to convert it to image source. I just added a method from nth-user's branch which seems to work fine and it avoids the memory isses. I also reduced the rendering size to the display size, which gave me additional 25% speed. Give it a try to the latest version.
